suite/spdz2k/src/main/java/dk/alexandra/fresco/suite/marlin/protocols/natives/Spdz2kAddKnownProtocol.java

```diff
@@ -0,0 +1,35 @@
+package dk.alexandra.fresco.suite.marlin.protocols.natives;
+
+import dk.alexandra.fresco.framework.DRes;
+import dk.alexandra.fresco.framework.network.Network;
+import dk.alexandra.fresco.framework.value.SInt;
+import dk.alexandra.fresco.suite.marlin.datatypes.CompUInt;
+import dk.alexandra.fresco.suite.marlin.datatypes.Spdz2kSInt;
+import dk.alexandra.fresco.suite.marlin.resource.Spdz2kResourcePool;
+
+public class Spdz2kAddKnownProtocol<PlainT extends CompUInt<?, ?, PlainT>>
+    extends Spdz2kNativeProtocol<SInt, PlainT> {
+
+  private final PlainT left;
+  private final DRes<SInt> right;
+  private SInt out;
+
+  public Spdz2kAddKnownProtocol(PlainT input, DRes<SInt> right) {
+    this.left = input;
+    this.right = right;
+  }
+
+  @Override
+  public EvaluationStatus evaluate(int round, Spdz2kResourcePool<PlainT> resourcePool,
+      Network network) {
+    out = ((Spdz2kSInt<PlainT>) right.out()).addConstant(left, resourcePool.getMyId(),
```

Comment:
Cast in the constructor, there is no more information available now

Reply:
Can't unfortunately because the result isn't available at construction time.

Reply:
The argument is quite a bit more sophisticated, but you are right